### PR TITLE
Updated the form fields

### PR DESCRIPTION
**Pull request by Sandhya1874**

### CODE REVIEW

File: server/views/pages/componentRequestForm.njk [hunk 1]
```diff
@@ -5,238 +5,278 @@
...
-      value: submittedForm.github_project_visibility,
-      label: {
-        text: "Github project visbility *"
-      },
-      items: [
-        {
-          value: "public",
-          text: "Public"
+        hint: {
+          text: "Seperated by comma without spaces eg: hmpps-developers,haha"
```
Comment (andrewrlee) [inline server/views/pages/componentRequestForm.njk:139]: little typo on these fields:`separated`

To be robust, we should really handle the possibility of spaces on our side and not worry the user about it. 
`teams.split(",").map(team => team.trim()).filter(team => Boolean(team))`

Which would: 
`"  ,   ,bob,,team  ,  some" => ["bob", "team", "some"]`

File: server/views/pages/componentRequestForm.njk [hunk 2]
```diff
@@ -5,238 +5,278 @@
...
+      }) }}
+      {{ govukRadios({
+        classes: "govuk-radios--inline",
+        idPrefix: "base_template",
+        name: "base_template",
+        value: submittedForm.base_template,
+        errorMessage: validationErrors | findError('base_template'),
+        fieldset: {
+          legend: {
+            text: "Github template repository *",
```
Comment (andrewrlee) [inline server/views/pages/componentRequestForm.njk:68]: we probably should support creating repos that aren't based on a template also.. maybe as a follow up?

File: server/views/pages/componentRequestForm.njk (hunk 1 above)
Comment (Sandhya1874) [reply server/views/pages/componentRequestForm.njk]: sorted as suggested

File: server/views/pages/componentRequestForm.njk (hunk 2 above)
Comment (Sandhya1874) [reply server/views/pages/componentRequestForm.njk:68]: Added github-only option

File: server/views/pages/componentRequestForm.njk [hunk 3]
```diff
@@ -5,238 +5,265 @@
...
-          text: "hmpps-template-kotlin"
+      }) }}
+      {{ govukInput({
+        id: "github_projects_teams_admin",
+        name: "github_projects_teams_admin",
+        value: submittedForm.github_projects_teams_admin,
+        errorMessage: validationErrors | findError('github_projects_teams_admin'),
+        classes: "govuk-!-width-two-third",
+        label: {
+          text: "Github Repository Admin Teams *",
```
Comment (james-jdgtl) [inline server/views/pages/componentRequestForm.njk:122]: Github repository Admin teams

File: server/views/pages/componentRequestForm.njk [hunk 4]
```diff
@@ -5,238 +5,265 @@
...
+        }
+      }) }}
+      {{ govukInput({
+        id: "github_project_teams_write",
+        name: "github_project_teams_write",
+        value: submittedForm.github_project_teams_write,
+        errorMessage: validationErrors | findError('github_project_teams_write'),
+        classes: "govuk-!-width-two-third",
+        label: {
+          text: "Github Repository Teams with write permissions",
```
Comment (james-jdgtl) [inline server/views/pages/componentRequestForm.njk:136]: Github repository teams with Write permissions

File: server/views/pages/componentRequestForm.njk [hunk 5]
```diff
@@ -5,238 +5,265 @@
...
+        }
+      }) }}
+      {{ govukInput({
+        id: "github_project_branch_protection_restricted_teams",
+        name: "github_project_branch_protection_restricted_teams",
+        value: submittedForm.github_project_branch_protection_restricted_teams,
+        errorMessage: validationErrors | findError('github_project_branch_protection_restricted_teams'),
+        classes: "govuk-!-width-two-third",
+        label: {
+          text: "Github Project Teams Branch protection restricted teams",
```
Comment (james-jdgtl) [inline server/views/pages/componentRequestForm.njk:150]: Github project branch protection restricted teams

(That's really clunky but we don't need two 'teams'.. maybe something better?)

File: server/views/pages/componentRequestForm.njk [hunk 6]
```diff
@@ -5,238 +5,265 @@
...
+            value: "github-only",
+            text: "github-only"
+          }]
+      }) }}
+      {{ govukSelect({
+        id: "github_project_visibility",
+        name: "github_project_visibility",
+        value: submittedForm.github_project_visibility,
+        label: {
+          text: "Github project visbility *",
```
Comment (james-jdgtl) [inline server/views/pages/componentRequestForm.njk:75]: visibility is missing an 'i'

File: server/views/pages/componentRequestForm.njk [hunk 7]
```diff
@@ -5,238 +5,265 @@
...
-      preventDoubleClick: true,
-      attributes: { "data-test": "submit" }
-    }) }}
+        id: "nonprod_alerts_severity_label",
+        name: "nonprod_alerts_severity_label",
+        value: submittedForm.nonprod_alerts_severity_label,
+        errorMessage: validationErrors | findError('nonprod_alerts_severity_label'),
+        classes: "govuk-!-width-one-third",
+        label: {
+          text: "Non-prod Severity label *",
```
Comment (james-jdgtl) [inline server/views/pages/componentRequestForm.njk:198]: Can we make these consistent?
Non-prod Alertmanager severity label
... maybe?

File: server/views/pages/componentRequestForm.njk [hunk 8]
```diff
@@ -5,238 +5,265 @@
...
+        }
+      }) }}
+        {{ govukInput({
+        id: "prod_alerts_severity_label",
+        name: "prod_alerts_severity_label",
+        value: submittedForm.prod_alerts_severity_label,
+        errorMessage: validationErrors | findError('prod_alerts_severity_label'),
+        classes: "govuk-!-width-one-third govuk-fieldset__legend--s",
+        label: {
+          text: "Production k8s alerts *",
```
Comment (james-jdgtl) [inline server/views/pages/componentRequestForm.njk:212]: Production Alertmanager severity label
...maybe?

File: server/views/pages/componentRequestForm.njk [hunk 9]
```diff
@@ -5,238 +5,265 @@
...
+        }
+      }) }}
+      {{ govukInput({
+        id: "requester_email",
+        name: "requester_email",
+        value: submittedForm.requester_email,
+        errorMessage: validationErrors | findError('requester_email'),
+        classes: "govuk-!-width-two-third govuk-fieldset__legend--s",
+        label: {
+          text: "Requester Email ID *",
```
Comment (james-jdgtl) [inline server/views/pages/componentRequestForm.njk:238]: Requester email ID (or address?)

File: server/views/pages/componentRequestForm.njk [hunk 10]
```diff
@@ -5,238 +5,265 @@
...
+      }) }}
+      <h3> Please Enter your contact details to send notification after processing request. </a></h3>
+      {{ govukInput({
+        id: "requester_name",
+        name: "requester_name",
+        value: submittedForm.requester_name,
+        errorMessage: validationErrors | findError('requester_name'),
+        classes: "govuk-!-width-one-half govuk-fieldset__legend--s",
+        label: {
+          text: "Requester Name *",
```
Comment (james-jdgtl) [inline server/views/pages/componentRequestForm.njk:227]: Requester name

File: server/views/pages/componentRequestForm.njk [hunk 11]
```diff
@@ -5,238 +5,265 @@
...
+        }
+      }) }}
+        {{ govukInput({
+        id: "slack_channel_nonprod_release_notify",
+        name: "slack_channel_nonprod_release_notify",
+        value: submittedForm.slack_channel_nonprod_release_notify,
+        errorMessage: validationErrors | findError('slack_channel_nonprod_release_notify'),
+        classes: "govuk-!-width-one-half",
+        label: {
+          text: "Slack channel for Non-Production release notifications *",
```
Comment (james-jdgtl) [inline server/views/pages/componentRequestForm.njk:175]: Slack channel for non-production release notifications *

File: server/views/pages/componentRequestForm.njk [hunk 12]
```diff
@@ -5,238 +5,265 @@
...
-        classes: "govuk-!-width-three-quarters"
-      }
-    }) }}
+        id: "slack_channel_prod_release_notify",
+        name: "slack_channel_prod_release_notify",
+        value: submittedForm.slack_channel_prod_release_notify,
+        errorMessage: validationErrors | findError('slack_channel_prod_release_notify'),
+        classes: "govuk-!-width-one-half",
+        label: {
+          text: "Slack channel for Production release notifications *",
```
Comment (james-jdgtl) [inline server/views/pages/componentRequestForm.njk:164]: Slack channel for production release notifications *

File: server/routes/forms.ts [hunk 13]
```diff
@@ -243,6 +243,21 @@ export default function routes({ componentNameService, serviceCatalogueService,
 }
 
 const buildFormData = (formData: Record<string, unknown>): GithubRepoRequestRequest => {
+  const githubProjectTeamsWrite = formData.github_project_teams_write
+    .toString()
```
Comment (psoleckimoj) [inline server/routes/forms.ts:247]: as this is done on 3 things, might be worth it being a function?

**Review by psoleckimoj — APPROVED** [review]
LGTM!

File: server/routes/forms.ts (hunk 13 above)
Comment (Sandhya1874) [reply server/routes/forms.ts]: replaced with function

**Pull request merged**